plugins/AudioFileProcessor/AudioFileProcessor.cpp

```diff
@@ -228,16 +228,18 @@ void AudioFileProcessor::loadSettings(const QDomElement& elem)
 	{
 		setAudioFile(elem.attribute("src"), false);
 
-		QString absolutePath = PathUtil::toAbsolute(m_sampleBuffer.audioFile());
+		QString absolutePath = PathUtil::toAbsolute(m_sample->sampleFile());
 		if (!QFileInfo(absolutePath).exists())
 		{
-			QString message = tr("Sample not found: %1").arg(m_sampleBuffer.audioFile());
+			QString message = tr("Sample not found: %1").arg(m_sample->sampleFile());
 			Engine::getSong()->collectError(message);
 		}
 	}
 	else if (!elem.attribute("sampledata").isEmpty())
 	{
-		m_sampleBuffer.loadFromBase64(elem.attribute("srcdata"));
+		auto buffer = gui::SampleLoader::createBufferFromBase64(elem.attribute("srcdata"));
+		// TODO C++20: Deprecated, use std::atomic<std::shared_ptr> instead
+		std::atomic_store(&m_sample, std::make_shared<Sample>(std::move(buffer)));
```

Comment:
You have a lot of code like this throughout the PR.

Maybe the `Sample` class could have a static method like this?
```c++
static void setBuffer(std::shared_ptr<Sample>* sample, std::unique_ptr<SampleBuffer> buffer)
{
	// TODO C++20: Deprecated, use std::atomic<std::shared_ptr> instead
	std::atomic_store(sample, std::make_shared<Sample>(std::move(buffer)));
} 
```

Then use it like:
```c++
Sample::setBuffer(&m_sample, gui::SampleLoader::createBufferFromBase64(elem.attribute("srcdata")));
```

Reply:
Discussed about this in another part of the review previously, but I think we can do away with ``std::atomic_load`` and ``std::atomic_store``. The control block of ``std::shared_ptr`` is thread-safe already, which I think is what we are really worried about here. Destruction of a shared pointer while being used by the audio thread would require another solution.